tests/Unit/ParallelAlgorithms/LinearSolver/Schwarz/Actions/Test_CommunicateOverlapFields.cpp

```diff
@@ -0,0 +1,192 @@
+      logging::Tags::Verbosity<DummyOptionsGroup>>{overlap,
+                                                   Verbosity::Verbose}};
+
+  // Setup element array
+  const auto add_element = [num_points_per_dim, overlap,
+                            &runner](const Element<Dim>& element) {
+    Mesh<Dim> mesh{num_points_per_dim, Spectral::Basis::Legendre,
+                   Spectral::Quadrature::GaussLobatto};
+    auto intruding_extents = make_array<Dim>(overlap);
+    typename fields_tag::type fields{mesh.number_of_grid_points()};
+    for (size_t i = 0; i < mesh.number_of_grid_points(); ++i) {
+      get(get<ScalarFieldTag<0>>(fields))[i] = i;
+      get(get<ScalarFieldTag<1>>(fields))[i] = i;
+    }
+    ActionTesting::emplace_component_and_initialize<element_array>(
+        make_not_null(&runner), element.id(),
+        {element, std::move(mesh), std::move(intruding_extents), size_t{0},
+         std::move(fields),
+         LinearSolver::Schwarz::OverlapMap<Dim, typename fields_tag::type>{}});
+  };
+  const ElementId<Dim> first_element_id{0};
+  const ElementId<Dim> second_element_id{1};
+  const ElementId<Dim> third_element_id{2};
+  // We can only have multiple face-neighbors in >1 dimensions
+  if constexpr (Dim > 1) {
+    add_element({first_element_id,
+                 {{Direction<Dim>::upper_xi(),
+                   {{second_element_id, third_element_id}, {}}}}});
+    add_element({second_element_id,
+                 {{Direction<Dim>::lower_xi(), {{first_element_id}, {}}}}});
+    add_element({third_element_id,
+                 {{Direction<Dim>::lower_xi(), {{first_element_id}, {}}}}});
+  } else {
+    add_element({first_element_id,
+                 {{Direction<Dim>::upper_xi(), {{second_element_id}, {}}}}});
+    add_element({second_element_id,
+                 {{Direction<Dim>::lower_xi(), {{first_element_id}, {}}}}});
+  }
+
+  ActionTesting::set_phase(make_not_null(&runner),
+                           metavariables::Phase::Testing);
+
+  // Send from first element
+  ActionTesting::next_action<element_array>(make_not_null(&runner),
+                                            first_element_id);
+  REQUIRE_FALSE(
+      ActionTesting::is_ready<element_array>(runner, first_element_id));
+  // Send from second element
+  ActionTesting::next_action<element_array>(make_not_null(&runner),
+                                            second_element_id);
+  REQUIRE(ActionTesting::is_ready<element_array>(runner, first_element_id) ==
+          (Dim == 1));
+  REQUIRE(ActionTesting::is_ready<element_array>(runner, second_element_id));
+  // Send from third element
+  if constexpr (Dim > 1) {
+    ActionTesting::next_action<element_array>(make_not_null(&runner),
+                                              third_element_id);
+    REQUIRE(ActionTesting::is_ready<element_array>(runner, first_element_id));
+    REQUIRE(ActionTesting::is_ready<element_array>(runner, third_element_id));
+  }
+  // Receive on first element
+  {
```

Comment:
I like scoping in tests, it prevents name conflicts. Instead of ending up with a test that has a bunch of variables like `expected_data_1` and `expected_data_2` that are only needed for a particular check, limiting their scope means the variables can just be named `expected_data`. This particular test doesn't strictly need the scope, but it doesn't hurt and may help in the future when extending it (I may want to add a similar check for another element and define the same `all_overlap_fields` variable. Then I don't have to refactor the test and add useless suffixes to the variables).